### PR TITLE
More builder validation and tests.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
Next up: map and any support. Then we ship!

File: src/main/java/com/squareup/protoparser/FieldElement.java [hunk 1]
```diff
@@ -114,8 +114,15 @@ public Builder addOption(OptionElement option) {
...
 
     public FieldElement build() {
+      checkNotNull(label, "label");
+      checkNotNull(type, "type");
+      checkNotNull(name, "name");
       checkNotNull(tag, "tag");
-      if (!isValidTag(tag)) throw new IllegalArgumentException("Illegal tag value: " + tag);
+
+      if (!isValidTag(tag)) {
+        throw new IllegalArgumentException("Illegal tag value: " + tag);
```
Comment (swankjesse) [inline src/main/java/com/squareup/protoparser/FieldElement.java]: checkArgument ?

Comment (swankjesse) [inline src/main/java/com/squareup/protoparser/FieldElement.java]: Ahh. . . this isn't guava.

**Comment by swankjesse** [comment]
LGTM

**Pull request merged**